src/vsr/sync.zig

```diff
@@ -0,0 +1,288 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const maybe = stdx.maybe;
+
+const vsr = @import("../vsr.zig");
+const constants = @import("../constants.zig");
+const stdx = @import("../stdx.zig");
+
+pub const SyncStage = union(enum) {
+    /// Not syncing.
+    none,
+
+    /// The commit lifecycle is in a stage that cannot be interrupted/cancelled.
+    /// We are waiting until that stage completes.
+    /// When it completes, we will terminate the commit and resume sync.
```

Comment:
Would be great to clarify "we will terminate the commit".

At first glance, it might surprise the reader, wondering how a commit that has completed, can be terminated (because it's completed, and because surely it's committed?).